modules/programs/home-manager.nix

```diff
@@ -13,10 +13,13 @@ in {
     programs.home-manager = {
       enable = mkEnableOption "Home Manager";
 
-      path = mkOption {
-        type = types.nullOr types.str;
-        default = null;
-        example = "$HOME/devel/home-manager";
+      paths = mkOption {
+        type = types.listOf types.str;
```

Comment:
Sounds like a good idea! Should I worry about the backwards compatibility of  `home-manager/default.nix` too? I could also attempt to preserve the default behavior of using `home-manager/default.nix` directly (while keeping the parameter name `path`).

Reply:
I agree about keeping the old name. I think it may also be a time to deprecate support for the

- `"\${XDG_CONFIG_HOME:-$HOME/.config}/nixpkgs/home-manager"`
- `"$HOME/.nixpkgs/home-manager"`

paths. I.e., keep this option entirely as-is. Similarly, keep `HOME_MANAGER_PATH`. Then make `home-manager/home-manager` error out with a helpful message if any of the deprecated paths are in play. Finally, remove the `for` loop and keep only the inner `if`.

In short, something like

``` bash
function setHomeManagerNixPath() {
    local path

    for p in "${XDG_CONFIG_HOME:-$HOME/.config}/nixpkgs/home-manager"
             "$HOME/.nixpkgs/home-manager"; do
        if [[ -e $p ]] ; then
            path="$p"
        fi
    done

    if [[ $path ]] ; then
        _i 'Helpful description saying that a deprecated path is used and how to resolve it'
        exit 1
    fi

    path="@HOME_MANAGER_PATH@"
    if [[ -e "$path" || "$path" =~ ^https?:// ]] ; then
        EXTRA_NIX_PATH+=("home-manager=$path")
    fi
}
```

Reply:
It feels weird for the script to fail because of something that exists on the user's system that we are not using anymore.

Maybe we should only warn the user if they do not provide a valid path and there is anything at the hard-coded paths that they are not telling home-manager to use explicitly.

We can add two warnings...
The first if the configured path is invalid

> Warning: home-manager missing from configured path /home/bob/bar/biz

The second warning would show, if we cant go with the configured path (if it was not provided or it is invalid), and for each of the old hard-coded paths that have directories that exists there.

> Warning: The fallback home-manager path "${XDG_CONFIG_HOME:-$HOME/.config}/nixpkgs/home-manager" has been deprecated and a file/directory was found there. To remove the warning, you can delete the file/directory at the deprecated path, or explicitly tell home-manager to use the path
>
> a) If you get home-manager from enabling it in your home-manager config, you can set this option in your config.
>
>        programs.home-manager.path = "\${XDG_CONFIG_HOME:-\$HOME/.config}/nixpkgs/home-manager"
>
>
> b) If you get home-manager from importing the code directly, use the `path` parameter when calling the home-manager package.
>
>        pkgs.callPackage /path/to/home-manager { path = "\${XDG_CONFIG_HOME:-\$HOME/.config}/nixpkgs/home-manager"; };